src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,195 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import duke.commands.ByeCommand;
+import duke.commands.Command;
+import duke.commands.DeadlineCommand;
+import duke.commands.DeleteCommand;
+import duke.commands.DoneCommand;
+import duke.commands.EventCommand;
+import duke.commands.FindCommand;
+import duke.commands.ListCommand;
+import duke.commands.SaveCommand;
+import duke.commands.TodoCommand;
+import duke.commands.UpdateCommand;
+import duke.exception.DukeException;
+import duke.tasks.DeadlineTask;
+import duke.tasks.EventTask;
+import duke.tasks.Task;
+import duke.tasks.TodoTask;
+
+/**
+ * Class that is responsible for parsing raw inputs and translating them to the correct Command or Task.
+ */
+public class Parser {
+
+    private static final int SINGLE_INPUT_COMMAND_SIZE = 1;
+    private static final int DOUBLE_INPUT_COMMAND_SIZE = 2;
+
+
+    /**
+     * Parses user input.
+     *
+     * @param input The user input.
+     * @return The Command subclass that correctly represents the user intent.
+     * @throws DukeException If input is unexpected or exceptional.
+     */
+    public static Command parseInput(String input) throws DukeException {
+        validateNotEmpty(input);
```

Comment:
Consider not making this method non-static and include a parser object as a field in Duke in order to call method